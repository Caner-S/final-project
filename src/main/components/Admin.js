import React from 'react';
import SpaceForm from "./SpaceForm";
import AllBookings from "./AllBookings";
import Grid from "@material-ui/core/Grid";
import AdminRequests from "./AdminRequests";

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import {Button} from "@material-ui/core";
import {updateSiteInformation} from "../dao/EnvironmentDao";

class Admin extends React.Component {
    constructor() {
        super();
        this.state = {
            businessCase: false,
        };
    }

    /*componentDidMount() {
        this.setState({businessCase: this.props.businessCase});
    }*/

    handleChange = (event) => {
        console.log(this.state.businessCase);
        this.setState({businessCase: event.target.value});
        console.log(this.state.businessCase);
    };

    printttt = e => {
        e.preventDefault();
        console.log("it works");
        console.log(this.state.businessCase);
        updateSiteInformation(this.state.businessCase);
    }


    render() {
        let {businessCase} = this.state;
        return (
            <div>
                <h1>Admin</h1>
                <SpaceForm/>
                <h4>COVID MODE</h4>
                <form onSubmit={this.printttt}>
                    <InputLabel >Require Business Case?</InputLabel>
                    <Select
                        id="demo-simple-select"
                        value={businessCase}
                        onChange={this.handleChange}
                    >
                        <MenuItem value={true}>Yes</MenuItem>
                        <MenuItem value={false}>No</MenuItem>
                    </Select>
                    <Button type="submit" >Submit</Button>
                </form>
                <Grid container>
                    <Grid item xs={12} s={12} m={6} zeroMinWidth>
                        <h4>All bookings</h4>
                        <AllBookings/>
                    </Grid>
                    <Grid item xs={12} s={12} m={6} zeroMinWidth>
                        <h4>All requests</h4>
                        <AdminRequests/>
                    </Grid>
                </Grid>
            </div>

        );
    }
}
export default Admin;