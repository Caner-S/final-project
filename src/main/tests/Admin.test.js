import React from 'react';
import renderer from 'react-test-renderer';

import Admin from "../components/Admin";

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<Admin />).toJSON();
    expect(tree).toMatchSnapshot();
});