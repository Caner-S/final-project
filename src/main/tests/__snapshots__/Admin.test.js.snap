// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <h1>
    Admin
  </h1>
  <div>
    <h4>
      New Office space
    </h4>
    <form
      onSubmit={[Function]}
    >
      <input
        name="number"
        onChange={[Function]}
        placeholder="Number"
        type="number"
        value={null}
      />
      <input
        name="capacity"
        onChange={[Function]}
        placeholder="Capacity"
        type="number"
        value={null}
      />
      <select
        name="type"
        onChange={[Function]}
        placeholder="Type"
        value={null}
      >
        <option
          value="Desk"
        >
          Desk
        </option>
        <option
          value="Meeting Room"
        >
          Meeting Room
        </option>
        <option
          value="Workspace"
        >
          Workspace
        </option>
      </select>
      <input
        name="exclusive"
        onChange={[Function]}
        placeholder="Exclusive"
        type="text"
        value={null}
      />
      <input
        name="section"
        onChange={[Function]}
        placeholder="Section"
        type="number"
        value={null}
      />
      <input
        name="floor"
        onChange={[Function]}
        placeholder="Floor"
        type="number"
        value={null}
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <h4>
    COVID MODE
  </h4>
  <form
    onSubmit={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
    >
      Require Business Case?
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="demo-simple-select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </form>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-12"
      m={6}
      s={12}
    >
      <h4>
        All bookings
      </h4>
      <div
        className="BookingsBorder"
      >
        <div
          className="MuiPaper-root MuiTableContainer-root container MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            aria-label="simple table"
            className="MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  User Id
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Space Number
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Arrival Time
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Departure Time
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
